### PR TITLE
Competition Dump Fixes

### LINKED ISSUE #369: Competition Dump fixes
Fix memory issue from dumping competition with big datasets by only outputting keys to datasets
(will address creating zip from stream, rather than having to load the entire dataset into memory 
in a separate issue: #370 )

fix incomplete dump by adding missing fields
- registration auto approve
- enable detailed results
- docker image
- queue (vhost)
- leaderboard: hidden
- column: hidden
- hide_output
- auto_migrate_to_this_phase
- description

fix terms not present

fix computation column indexes appearing as string rather than list

### LINKED ISSUE #379: Add description to competition unpackers
Description field is not present in the unpacking step of competitions, should be added to both the v1 and v2 unpackers

### PR DESCRIPTION
**Pull request by jimmykodes**
# @ mention of reviewers
@ckcollab 

# A brief description of the purpose of the changes contained in this PR.
adding fixes so that a re-uploaded dump will create the same competition it was dumped from. Add the appropriate fields for this, and also output dataset keys instead of zipping the entire dataset, cause a memory exceeded error from celery

# Issues this PR resolves
resolves #369 
resolves #379 

# Known issues to be addressed in a separate PR
Does not address the contents of #370 

# A checklist for hand testing
- [ ] upload a competition of your choosing
- [ ] create a dump of that competition
- [ ] download and re-upload the dump zip and verify the competition is the same

# Checklist
- [x] Code review by me 
- [x] Hand tested by me 
- [x] I'm proud of my work
- [x] Code review by reviewer
- [x] Hand tested by reviewer
- [x] Ready to merge

### CODE REVIEW

**Comment by ckcollab** [comment]
Made https://github.com/codalab/competitions-v2/issues/373 to allow switching between modes (assuming we can't do this already, _pretty sure_ we can't). Would be dependent on #370.

File: src/apps/competitions/tests/unpacker_test_data.py [hunk 1]
```diff
@@ -200,6 +200,39 @@
     }
 ]
 
+PHASES2 = [
```
Comment (jimmykodes) [inline src/apps/competitions/tests/unpacker_test_data.py]: I hate this solution, but i wasn't sure what the best way to handle the fact that Phases for v2 has `hide_output` and v1.5 does not. There is probably a better solution than this, I just can't think of it.

File: src/apps/competitions/tests/test_v15_unpacker.py [hunk 2]
```diff
@@ -25,7 +25,7 @@ def test_task_unpacking(self):
 
     def test_phase_unpacking(self):
         self.unpacker._unpack_phases()
-        assert self.unpacker.competition['phases'] == test_data.PHASES
+        assert self.unpacker.competition['phases'] == test_data.PHASES1
```
Comment (ckcollab) [inline src/apps/competitions/tests/test_v15_unpacker.py]: could name it `PHASES_v15` `PHASES_v2` ... put _some_ lipstick on the pig? :)

File: src/apps/competitions/tests/unpacker_test_data.py [hunk 3]
```diff
@@ -200,6 +200,24 @@
     }
 ]
 
+V2_SPECIFIC_PHASE_DATA = [
+    # Tuples of (key, value) of data specific to v2 unpacker.
+    ('hide_output', False)
+]
+
+
+def get_phases(version):
```
Comment (jimmykodes) [inline src/apps/competitions/tests/unpacker_test_data.py]: @ckcollab thoughts on this fix instead?

File: src/apps/competitions/unpackers/base_unpacker.py [hunk 4]
```diff
@@ -263,6 +263,7 @@ def _save_tasks(self):
                         continue
                     key, temp_data_path = self._get_data_key(**task_file_data)
                     task[file_type] = key
+                # TODO: Pass created_by explicitly here instead of request
```
Comment (jimmykodes) [inline src/apps/competitions/unpackers/base_unpacker.py]: Should I address this in this PR? or create a new issue for later?

File: src/apps/competitions/tests/unpacker_test_data.py (hunk 3 above)
Comment (ckcollab) [reply src/apps/competitions/tests/unpacker_test_data.py]: cool ty!

**Comment by ckcollab** [comment]
:shipit: just tested (pre develop merge) and things are fine

**Pull request merged**